pages/api/submit.tsx

```diff
@@ -117,28 +117,29 @@ const processFormData = async (
       return await setTimeout(() => res.status(200).json({ received: true }), 1000);
     }
 
-    for (const [key, value] of Object.entries(files)) {
+    for (const [_key, value] of Object.entries(files)) {
       const fileOrArray = value;
       if (!Array.isArray(fileOrArray)) {
         if (fileOrArray.name) {
-          const { isValid, result } = await pushFileToS3(fileOrArray, fileOrArray.name);
-          if (isValid) reqFields[key] = result;
-          else throw new Error(result as string);
+          logMessage.info(`uploading: ${_key} - filename ${fileOrArray.name} `);
+          const { isValid, key } = await pushFileToS3(fileOrArray);
+          if (isValid) {
+            uploadedFilesKeyUrlMapping.set(fileOrArray.name, key);
+            reqFields[_key] = key;
+          }
         }
       } else if (Array.isArray(fileOrArray)) {
-        //TODO when we start uploading more than a file this code should be refactored.
-        fileOrArray.forEach(async (fileItem) => {
+        // An array will be returned in a field that includes multiple files
+        fileOrArray.forEach(async (fileItem, index) => {
           if (fileItem.name) {
-            const { isValid, result } = await pushFileToS3(fileItem, fileItem.name);
-            //TODO @bryan-robitaille.
-            if (isValid)
-              /** what's the best approach to store these links in response object*/
-              // Need to check to see if the below is easy to parse out when processing the form submission
-              reqFields[key] = result;
-            else throw new Error(result);
+            logMessage.info(`uploading: ${_key} - filename ${fileItem.name} `);
+            const { isValid, key } = await pushFileToS3(fileItem);
+            if (isValid) {
+              uploadedFilesKeyUrlMapping.set(fileItem.name, key);
+              reqFields[`${_key}-${index}`] = key;
```

Comment:
Added return value (file path) to the reqForms form field property otherwise the value would never be passed to the submission lambda function.